### PR TITLE
Only split user agent if it can be split

### LINKED ISSUE #94: System.ArgumentOutOfRangeException: length ('-1') must be a non-negative value. (Parameter 'length') in GetDescription
I've been getting the following issue after updating the DLNA plugin.  I'm guessing that not all UserAgent's contain
a `/` (especially for embedded devices/TVs).

Think the issue is here:

https://github.com/jellyfin/jellyfin-plugin-dlna/blob/edcdf04acd36c2992bf42446bdef1da54e4e5e53/src/Jellyfin.Plugin.Dlna/Api/DlnaServerController.cs#L67-L71

Here's my stacktrace.

```
[2024-11-04 14:29:15.909 +00:00] [ERR] Error processing request. URL "GET" "/dlna/XXXX/description.xml".
System.ArgumentOutOfRangeException: length ('-1') must be a non-negative value. (Parameter 'length')
Actual value was -1.
   at System.ArgumentOutOfRangeException.ThrowNegative[T](T value, String paramName)
   at System.ArgumentOutOfRangeException.ThrowIfNegative[T](T value, String paramName)
   at System.String.ThrowSubstringArgumentOutOfRange(Int32 startIndex, Int32 length)
   at Jellyfin.Plugin.Dlna.Api.DlnaServerController.GetDescriptionXml(String serverId)
   at lambda_method1570(Closure, Object, Object[])
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Jellyfin.Api.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Api.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Api.Middleware.IPBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Api.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Api.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Api.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Api.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
   at Jellyfin.Api.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

### PR DESCRIPTION
**Pull request by Shadowghost**
Fixes #94

### CODE REVIEW

File: src/Jellyfin.Plugin.Dlna/Api/DlnaServerController.cs [hunk 1]
```diff
@@ -66,7 +66,12 @@ public ActionResult<string> GetDescriptionXml([FromRoute, Required] string serve
         string? userAgent = Request.Headers.UserAgent;
         if (userAgent is not null)
         {
-            userAgent = userAgent.Substring(0, userAgent.IndexOf('/'));
+            var firstIndexOfSlash = userAgent.IndexOf('/');
+            if (firstIndexOfSlash > 0)
+            {
+                userAgent = userAgent.Substring(0, firstIndexOfSlash);
```
Comment (crobibero) [inline src/Jellyfin.Plugin.Dlna/Api/DlnaServerController.cs:72]: Can this be changed to just use our `LeftPart` function instead?

Comment (Shadowghost) [reply src/Jellyfin.Plugin.Dlna/Api/DlnaServerController.cs:72]: I don't think this is available in a plugin 🤔

Comment (Shadowghost) [reply src/Jellyfin.Plugin.Dlna/Api/DlnaServerController.cs:72]: Not sure if it's worth pulling in that package just to convert a string to a span, use the extension function and then convert back to a string

**Review by crobibero — APPROVED** [review]

**Comment by miegl** [comment]
Please make a release with this bug fix, many people are hitting the bug and it unnecessarily slows adoption of Jellyfin. Thank you!

**Pull request merged**